### PR TITLE
Use testdb in integration tests

### PR DESCRIPTION
**Pull request by gdbelvin**
Use the KT test database which sets the right flags.

Don't use the same flag for the database as Trillian.

### CODE REVIEW

File: impl/integration/env.go [hunk 1]
```diff
@@ -115,14 +116,11 @@ func keyFromPEM(p string) *any.Any {
 }
 
 // NewEnv sets up common resources for tests.
-func NewEnv(ctx context.Context) (*Env, error) {
+func NewEnv(ctx context.Context, t testing.TB) (*Env, error) {
```
Comment (mhutchinson) [inline impl/integration/env.go]: Should `t.Helper()` be called here?

File: impl/sql/testdb/testdb.go [hunk 2]
```diff
@@ -27,12 +26,13 @@ import (
 	ktsql "github.com/google/keytransparency/impl/sql"
 )
 
-var dataSourceURI = flag.String("test_mysql_uri", "root@tcp(127.0.0.1)/", "The MySQL URI to use when running tests")
+var dataSourceURI = "root@tcp(127.0.0.1)/" //  "The MySQL URI to use when running tests"
```
Comment (mhutchinson) [inline impl/sql/testdb/testdb.go]: Is it really desired to not have any flag control over this value?

Comment (gdbelvin) [reply impl/sql/testdb/testdb.go]: Options:
1) Have a KT specific flag
2) Export the flag that Trillian defines

Comment (gdbelvin) [reply impl/sql/testdb/testdb.go]: Going for option 1

File: impl/sql/testdb/testdb.go [hunk 3]
```diff
@@ -27,11 +27,12 @@ import (
 	ktsql "github.com/google/keytransparency/impl/sql"
 )
 
-var dataSourceURI = flag.String("test_mysql_uri", "root@tcp(127.0.0.1)/", "The MySQL URI to use when running tests")
+var dataSourceURI = flag.String("test_kt_uri", "root@tcp(127.0.0.1)/", "The MySQL URI to use when running tests")
```
Comment (mhutchinson) [inline impl/sql/testdb/testdb.go]: flag should have `mysql` or `db` or something in it.

**Pull request merged**